src/django_upgrade/fixers/request_headers.py

```diff
@@ -58,6 +58,44 @@ def visit_Call(
         )
 
 
+@fixer.register(ast.In)
+def visit_In(
+    state: State,
+    node: ast.In,
+    parents: list[ast.AST],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        type(parents[-1]) is ast.Compare
+        and len(parents[-1].comparators) == 1
+        and is_request_or_self_request_meta(parents[-1].comparators[0])
+        and type(parents[-1].left) is ast.Constant
+        and (raw_header_name := get_http_header_name(parents[-1].left.value))
+        is not None
+    ):
+        yield ast_start_offset(parents[-1]), partial(
+            rewrite_in_statement, raw_header_name=raw_header_name
+        )
+
+
+@fixer.register(ast.NotIn)
+def visit_NotIn(
+    state: State,
+    node: ast.NotIn,
+    parents: list[ast.AST],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        type(parents[-1]) is ast.Compare
+        and len(parents[-1].comparators) == 1
+        and is_request_or_self_request_meta(parents[-1].comparators[0])
+        and type(parents[-1].left) is ast.Constant
+        and (raw_header_name := get_http_header_name(parents[-1].left.value))
+        is not None
+    ):
+        yield ast_start_offset(parents[-1]), partial(
+            rewrite_in_statement, raw_header_name=raw_header_name
+        )
```

Comment:
if you change as above to a `visit_Compare` function, this can be merged in, just using `isinstance(node.ops[0], (ast.In, ast.NotIn))`